modules/gke-cluster/main.tf

```diff
@@ -59,8 +59,11 @@ resource "google_container_cluster" "cluster" {
 
   # TODO(ludomagno): compute addons map in locals and use a single dynamic block
   addons_config {
-    dns_cache_config {
-      enabled = var.addons.dns_cache_config
+    dynamic "dns_cache_config" {
+      for_each = var.enable_autopilot == true ? [] : [""]
```

Comment:
Can you change the condition to just `var.enable_autopilot`?

Reply:
Done, thanks Julio!